src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,6 +1256,794 @@ subroutine eigencalving(domain, err)
+                    realArgs=(/minval(meanFlowParamA), maxval(meanFlowParamA)/))
+         if (config_damage_use_constant_A) then
+             where (thickness == 0.0_RKIND)
+                 visc = 0.0_RKIND
+             elsewhere
+                 visc = 0.5*stiffnessFactor*config_default_flowParamA**(-1.0/3.0)*(sqrt(exx**2+eyy**2 + exx*eyy + 0.25*(exy+eyx)**2))**((1.0-3.0)/3.0)
+             endwhere
+         else
+             where (thickness == 0.0_RKIND)
+                 visc = 0.0_RKIND
+             elsewhere
+                 visc = 0.5*stiffnessFactor*meanFlowParamA**(-1.0/3.0)*(sqrt(exx**2+eyy**2 + exx*eyy + 0.25*(exy+eyx)**2))**((1.0-3.0)/3.0)
+             endwhere
+         end if
+         ! recover effective viscosity from vel. and temp. fields
+
+         sigmaxy(:) = 2*visc(:)*(exy(:)+eyx(:))/2.0
+         sigmayx(:) = 2*visc(:)*(eyx(:)+exy(:))/2.0
+
+         sigmaxx(:) = 2*visc(:)*exx(:)
+         sigmayy(:) = 2*visc(:)*eyy(:)
+         !deviatoric stress
+
+         sigmamin(:) = (sigmaxx(:)+sigmayy(:))/2.0-sqrt(((sigmaxx(:)-sigmayy(:))/2.0)**2+((sigmaxy(:)+sigmayx(:))/2)**2)
+         sigmamax(:) = (sigmaxx(:)+sigmayy(:))/2.0+sqrt(((sigmaxx(:)-sigmayy(:))/2.0)**2+((sigmaxy(:)+sigmayx(:))/2)**2)
+         !deviatoric principal stress
+
+         where (thickness == 0.0_RKIND)
+             alpha = 0.0_RKIND
+             epsmax = 0.0_RKIND
+             szero = 0.0_RKIND
+         elsewhere
+             alpha = sigmamin / sigmamax
+             epsmax = sigmamax / (2.0d0*visc)
+             szero = rhoi*(rhoo-rhoi)*gravity*thickness/(2.0d0*sigmamax*rhoo)
+         endwhere
+             
+         !alpha(:) = sigmamin(:) / (sigmamax(:))
+         !epsmax(:) = sigmamax(:) / (2.0d0*visc(:))
+         !szero(:) = rhoi*(rhoo-rhoi)*gravity*thickness(:)/(2.0d0*sigmamax(:)*rhoo)
+          ! Compute the hydrostatic to tensile stress ratio
+          ! These commented lines are doing the same thing as the above where loop. Which is better?
+
+         nstar(:) = 12.0d0*(1.0d0+alpha(:)+alpha(:)**2)/(4.0d0*(1.0d0+alpha(:)+alpha(:)**2)+6.0d0*alpha(:)**2) 
+          ! Compute the effective flow law exponent
+
+         !szero(:) = rhoi*(rhoo-rhoi)*gravity*thickness(:)/(2.0d0*sigmamax(:)*rhoo)
+          ! Compute the hydrostatic to tensile stress ratio
+
+         !do iCell = 1, nCells
+         !    if (li_mask_is_floating_ice(cellMask(iCell))) then
+         !        floatingBasalMassBal(iCell) = 0.2/31536000.0*tanh((lowerSurface(iCell)-bedTopography(iCell))/75.0) * max(-100-lowerSurface(iCell),0.0)
+         !    end if
+         !end do
+         ! Temporary change of BMB for the MISMIP+ Ice1r experiment
+
+         do iCell = 1, nCells
+         if (thickness(iCell) == 0.0_RKIND) then
+             source(iCell) = 0.0_RKIND
+         else
+             source(iCell) = nstar(iCell) * (1.0d0-szero(iCell))*epsmax(iCell)-floatingBasalMassBal(iCell)/rhoi/(thickness(iCell))
```

Comment:
Any numbers should have a suffix of `_RKIND`appended to them.

Reply:
added